src/table.rs

```diff
@@ -10,15 +10,16 @@ use crate::{InlineTable, InternalString, Item, KeyMut, Value};
 /// Type representing a TOML non-inline table
 #[derive(Clone, Debug, Default)]
 pub struct Table {
-    // comments/spaces before and after the header
+    // Comments/spaces before and after the header
     pub(crate) decor: Decor,
-    // whether to hide an empty table
+    // Whether to hide an empty table
     pub(crate) implicit: bool,
-    // whether this is a proxy for dotted keys
+    // Whether this is a proxy for dotted keys
     pub(crate) dotted: bool,
-    // used for putting tables back in their original order when serialising.
-    // Will be None when the Table wasn't parsed from a file.
-    pub(crate) position: Option<usize>,
+    // Used for putting tables back in their original order when serialising.
+    //
+    // `None` for user created tables (can be overridden with `set_position`)
+    doc_position: Option<usize>,
```

Comment:
Is this position always in a `Document`? Or is this used for nested tables as well? Would `parent_position` as in "position in the parent" be a better name?

Reply:
No matter the nesting level, its is the global position in the document or otherwise we wouldn't be able to preserve cases like
```toml
[foo.bar]

[baz]

[foo.bar.yum]

[foo]
```